src/main/kotlin/io/titandata/titan/providers/local/Checkout.kt

```diff
@@ -5,21 +5,44 @@
 package io.titandata.titan.providers.local
 
 import io.titandata.client.apis.CommitsApi
+import io.titandata.client.apis.RepositoriesApi
 import io.titandata.titan.clients.Docker
 import io.titandata.titan.utils.CommandExecutor
 
 class Checkout (
         private val commandExecutor: CommandExecutor = CommandExecutor(),
         private val docker: Docker = Docker(commandExecutor),
-        private val commitsApi: CommitsApi = CommitsApi()
+        private val commitsApi: CommitsApi = CommitsApi(),
+        private val repositoriesApi: RepositoriesApi = RepositoriesApi()
 ) {
-    fun checkout(container: String, guid: String) {
+    fun checkout(container: String, commit: String?, tags: List<String>) {
+        var sourceCommit = if (commit == null) {
```

Comment:
is sourceCommit ever altered? it looks immutable with the if statements. Should it be `val sourceCommit`?